### PR TITLE
insert null into Integer column using JDBC

### PR DESCRIPTION
**Pull request by mladensavic94**

### CODE REVIEW

File: persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class DBConfig {
+
+    private static Connection INSTANCE;
+
+    public static Connection getConnection() throws Exception {
+        if (INSTANCE == null) {
+            Class.forName("org.h2.Driver");
```
Comment (JonCook) [inline persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java:14]: Do we need this Class.forName - looks redundant?

File: persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class DBConfig {
+
+    private static Connection INSTANCE;
+
+    public static Connection getConnection() throws Exception {
```
Comment (JonCook) [inline persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java:12]: Can we throw the proper Exception type here? SQLException

File: persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        if (INSTANCE == null) {
+            Class.forName("org.h2.Driver");
+            INSTANCE = DriverManager.getConnection("jdbc:h2:mem:insertnull", "user", "password");
+            createPersonTable();
+        }
+        return INSTANCE;
+    }
+
+    private static void createPersonTable() {
```
Comment (JonCook) [inline persistence-modules/core-java-persistence-2/src/main/java/com/baeldung/insertnull/DBConfig.java:21]: Let's use try-with resources in this method and get rid of the printStacktrace

File: persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package com.baeldung.insertnull;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
```
Comment (JonCook) [inline persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java:4]: Let's please remove the unused import statements.

File: persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            preparedStatement.setString(2, person.getName());
+            preparedStatement.setString(3, person.getLastName());
+            if (person.getAge() == null)
+                preparedStatement.setNull(4, Types.INTEGER);
+            else
+                preparedStatement.setInt(4, person.getAge());
+            int noOfRows = preparedStatement.executeUpdate();
+
+            assertThat(noOfRows, equalTo(1));
+        } catch (Exception ignored) {
```
Comment (JonCook) [inline persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java:33]: Why do we need these catch Exception statements if they don't do anything and the method already throws Exception. let's remove them from all the tests.

**Review by JonCook — CHANGES_REQUESTED** [review]
Hi @mladensavic94 - a few things to clean up there. Thanks.

File: persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    public void givenNewPerson_whenSetNullIsUsed_thenNewRecordIsCreated() throws SQLException {
+        Person person = new Person(1, "John", "Doe", null);
+
+        try (PreparedStatement preparedStatement = DBConfig.getConnection().prepareStatement(SQL)) {
+            preparedStatement.setInt(1, person.getId());
+            preparedStatement.setString(2, person.getName());
+            preparedStatement.setString(3, person.getLastName());
+            if (person.getAge() == null)
```
Comment (eric-martin) [inline persistence-modules/core-java-persistence-2/src/test/java/com/baeldung/insertnull/InsertNullUnitTest.java:24]: Add braces for one line if/else statements

**Pull request merged**